### DISCUSSION TITLE
queryFn, call different endpoint every Nth repetition

### DISCUSSION THREAD

**Discussion by czescwojtek** [TanStack/query#2833] [Q&A] [answered]
Is it possible to use useQuery hook with refetchInterval option set to run different queryFn every nth, let's say 10th, repetition like so:
1. calls /false endpoint, 2. ... 9. calls /false endpoint, 10. calls /true endpoint, 11. calls /false endpoint
... and so on

**Comment by TkDodo** [comment, marked as answer]
You would need to do that in userland - keeping track of the amount of fetches in local state, then changing the url your queryFn goes to depending on that.

**Reply by czescwojtek** [reply]
What if the hook is being used several times across whole app? I've followed your suggestion and introduced a 'recoil' atom which keeps the track of the amount of fetches (updated by useQuery 'onSettled') but due to the fact the hook is being used multiple times the 'onSettled' fires X times.

**Reply by TkDodo** [reply]
I guess you can use the `onSuccess` and `onError` callbacks on the queryCache, which will only be invoked once.
